torch/amp/grad_scaler.py

```diff
@@ -281,6 +281,15 @@ def _unscale_grads_(
                         per_device_found_inf.get(device),
                         per_device_inv_scale.get(device),
                     )
+                    if "DTensor" in grads[0].__class__.__name__:
+                        reduce_result = found_inf
```

Comment:
does it make a difference with/without `reduce_result = found_inf` ? because DTensor always have full_tensor method, seems `reduce_result = max_tensor.full_tensor()` is always executed